app/src/ui/lib/authentication-form.tsx

```diff
@@ -26,7 +26,7 @@ interface IAuthenticationFormProps {
   readonly onBrowserSignInRequested: () => void
 
   /** An array of additional buttons to render after the "Sign In" button. */
-  readonly additionalButtons?: ReadonlyArray<JSX.Element>
+  readonly additionalButtons?: ReadonlyArray<JSX.Element> | number | string | boolean | {} | null
```

Comment:
So I think we should leave these typed as `ReadonlyArray<JSX.Element>` and deal with React's totally pants `children` typing at the callsite by casting. That way we're protecting our consumers from having to deal with this hot mess.